src/commands/moderation/giverole.js

```diff
@@ -0,0 +1,35 @@
+const { Command, CommandError, Constants, SwitchbladeEmbed } = require('../../')
+
+module.exports = class Giverole extends Command {
+  constructor (client) {
+    super(client, {
+      name: 'giverole',
+      category: 'moderation',
+      requirements: { guildOnly: true, botPermissions: ['MANAGE_ROLES'], permissions: ['MANAGE_ROLES'] },
+      parameters: [{
+        type: 'member', acceptBot: true, missingError: 'commands:giverole.missingUser'
+      }, {
+        type: 'role', full: true, missingError: 'errors:invalidRole'
+      }]
+    })
+  }
+
+  async run ({ channel, member, author, t }, targetMember, role) {
+    const embed = new SwitchbladeEmbed(author)
+    await targetMember.addRole(role).then(modifiedMember => {
+      if (!member.hasPermission('ADMINISTRATOR', false, true, false)) {
+        if (member.highestRole.comparePositionTo(role) <= 0) throw new CommandError(t('commands:giverole.roleAboveAuthors'))
+        if (member.highestRole.comparePositionTo(member.highestRole) <= 0) throw new CommandError(t('commands:giverole.roleBelow'))
+      }
```

Comment:
This verification process should be made before adding the role to the target member.

And the condition in line 22 is non-sense, you are comparing `member.highestRole` to itself: **`member.highestRole`**`.comparePositionTo(`**`member.highestRole`**`)`

Reply:
Oh shit, I just changed the variable names and forgot this one lol.